build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -974,4 +974,81 @@ export const handlers: Handler[] = [
 			}
 		},
 	},
+	{
+		name: "npm-package-json-clean-script",
+		match,
+		handler: (file, root) => {
+			// This rule enforces that we have a module field in the package iff we have a ESM build
+			// So that tools like webpack will pack up the right version.
+			let json;
+
+			try {
+				json = JSON.parse(readFile(file));
+			} catch (err) {
+				return "Error parsing JSON file: " + file;
+			}
+
+			const scripts = json.scripts;
+			if (scripts === undefined) {
+				return undefined;
+			}
+
+			const missing = missingCleanDirectories(scripts);
+
+			if (missing.length !== 0) {
+				return `Clean script missing the following:${missing
+					.map((i) => `\n\t${i}`)
+					.join("")}`;
+			}
+		},
+		resolver: (file, root) => {
+			updatePackageJsonFile(path.dirname(file), (json) => {
+				let missing = missingCleanDirectories(json.scripts);
+				if (missing.length === 0) {
+					return;
+				}
+				const clean = json.scripts["clean"] ?? "rimraf --glob";
+				if (clean.includes('"')) {
+					missing = missing.map((name) => `"${name}"`);
+				}
```

Comment:
I was thinking of using single quotes if possible, so we don't have to escape them, but also to _always_ use them. I see some scripts are as simple as `rimraf --glob nyc` and technically don't need quotes, but using them every time would make it harder for someone to add an unquoted pattern with an asterisk which might not work on Windows.

Reply:
Done.